src/freertos_drivers/net_cc32xx/CC32xxWiFi.cxx

```diff
@@ -542,6 +544,31 @@ void CC32xxWiFi::wlan_connect(const char *ssid, const char* security_key,
     }
 }
 
+/*
+ * CC32xxWiFi::wlan_wps_pbc_initiate()
+ */
+int CC32xxWiFi::wlan_wps_pbc_initiate()
+{
+    SlSecParams_t sec_params;
+    sec_params.Key = (signed char*)"";
+    sec_params.KeyLen = 0;
+    sec_params.Type = SL_WLAN_SEC_TYPE_WPS_PBC;
+
+    int result = sl_WlanConnect((signed char*)"WPS_AP", strlen("WPS_AP"), NULL,
+                                &sec_params, NULL);
+    HASSERT(result >= 0);
+
+    sleep(2);
+
+    for (int i = 0; i < 20 && !wlan_ready(); ++i)
+    {
+        connecting_update_blinker();
```

Comment:
i think it would be a better API to have this busyloop be the responsibility of the caller. It's not clear whether we will be using the blinker to show the connection progress; on the throttle for example the LCD would be better. If we don't have any of the sleep commands in here, we can choose not to start a new thread for the WPS but call this directly from the executor or network thread.